src/AbstractFactory.php

```diff
@@ -83,7 +84,7 @@ protected function retrieveConfig(ContainerInterface $container, string $configK
         $sectionConfig     = $applicationConfig['doctrine'][$section] ?? [];
 
         if (array_key_exists($configKey, $sectionConfig)) {
-            return $sectionConfig[$configKey] + $this->getDefaultConfig($configKey);
+            return array_replace_recursive($this->getDefaultConfig($configKey), $sectionConfig[$configKey]);
```

Comment:
This is actually a bugfix which could also be passed to lower versions of this component.
I've written down some more details in the commit message:

> In `Roave\PsrContainerDoctrine\ConfigurationFactory`, the `doctrine/orm` configuration is assembled. That configuration has a nested array configuration for `second_level_cache` which also have defaults. 
> Due to the array concatenation, nested array values are not applied and thus the configured defaults within the `ConfigurationFactory` are not applied.

Reply:
Depending on how critical this is, we may backport it, but since nobody cried yet: :shrug: 

This will be released in a new minor anyway, so upgrading should be trivial.

Reply:
Yah, would say the same. If someone wants this in 3.3.x to make it work on older doctrine/* component versions (which we dropped in this patch) then this can be easily backported.

Reply:
See https://github.com/Roave/psr-container-doctrine/pull/65

Reply:
This breaks configuration for `doctrine/migrations`. It's not possible to drop the default `migrations_paths`.

Reply:
@yethee I just released `3.4.1` with @Slamdunk's suggestion.